### PR TITLE
users wip

### PR DESCRIPTION
**Pull request by ianhoffman**

### CODE REVIEW

File: app/airq/models/users.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from flask_login import UserMixin
+from werkzeug.security import check_password_hash
+from werkzeug.security import generate_password_hash
```
Comment (emjgreen) [inline app/airq/models/users.py:3]: Why did you choose werkzeug? Just curious

File: app/airq/templates/base.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<!doctype html>
+<title>{% block title %}{% endblock %} - Hazebot</title>
+<nav>
+  <h1>Hazebot</h1>
+  <ul>
+    {% if g.user %}
+      <li><span>{{ g.user['username'] }}</span>
+      <li><a href="{{ url_for('logout') }}">Log Out</a>
```
Comment (emjgreen) [inline app/airq/templates/base.html:8]: Stupid flask!

**Review by emjgreen — DISMISSED** [review]

File: app/airq/models/users.py (hunk 1 above)
Comment (ianhoffman) [reply app/airq/models/users.py:3]: It's the default for Flask. I hate that flask has so many dependencies and you actually end up needing most of them.

File: app/airq/templates/base.html (hunk 2 above)
Comment (ianhoffman) [reply app/airq/templates/base.html:8]: yeah, this is JINJA

File: app/airq/__init__.py [hunk 3]
```diff
@@ -12,5 +12,6 @@
 
 # Register routes
 app.route("/", methods=["GET"])(api.healthcheck)
+app.route("/login", methods=["GET", "POST"])(api.login)
+app.route("/test", methods=["GET"])(api.test_command)
 app.route("/sms", methods=["POST"])(api.sms_reply)
```
Comment (emjgreen) [inline app/airq/__init__.py:17]: Nice

File: app/airq/models/clients.py [hunk 4]
```diff
@@ -197,3 +207,45 @@ def maybe_notify(self) -> bool:
...
+                        (
+                            cls.last_activity_at > curr_time - (window * 24 * 60 * 60),
+                            literal_column("1"),
+                        )
+                    ],
+                    else_=literal_column("0"),
+                )
+            ).label("{} day{}".format(window, "s" if window > 1 else ""))
+            for window in windows
+        ]
```
Comment (emjgreen) [inline app/airq/models/clients.py:243]: What is this "case, when" stuff, and "literal_column" ?

**Review by emjgreen — DISMISSED** [review]
LGTM, Still getting used to SQLalchemy

File: app/airq/models/clients.py (hunk 4 above)
Comment (ianhoffman) [reply app/airq/models/clients.py:264]: It constructs SQL like the following:

```
SELECT 
   SUM(CASE WHEN condition_1 THEN 1 ELSE 0 END) as label_1,
   SUM(CASE WHEN condition_2 THEN 1 ELSE 0 END) as label_2,
   ... etc ..
FROM ..
```
So it lets us do a bunch of different counts in one query, basically.

**Pull request merged**